src/sidebar/util/test/annotation-sharing-test.js

```diff
@@ -24,75 +25,120 @@ describe('sidebar.util.annotation-sharing', () => {
 
-    it('returns true if service settings `enableShareLinks` is non-boolean', () => {
-      fakeServiceConfig.returns({ enableShareLinks: 'foo' });
-      assert.isTrue(sharingUtil.sharingEnabled({}));
-    });
+        assert.equal(
+          sharingUtil.annotationSharingLink(fakeAnnotation),
+          'https://www.example2.com'
+        );
+      });
 
-    it('returns false if service settings really sets it to nope', () => {
-      fakeServiceConfig.returns({ enableShareLinks: false });
-      assert.isFalse(sharingUtil.sharingEnabled({}));
-    });
-  });
+      it('returns `undefined` if links empty', () => {
+        delete fakeAnnotation.links.incontext;
+        delete fakeAnnotation.links.html;
 
-  describe('#shareURI', () => {
-    it('returns `incontext` link if set on annotation', () => {
-      assert.equal(
-        sharingUtil.shareURI(fakeAnnotation),
-        'https://www.example.com'
-      );
+        assert.isUndefined(sharingUtil.annotationSharingLink(fakeAnnotation));
+      });
+
+      it('returns `undefined` if no links on annotation', () => {
+        delete fakeAnnotation.links;
+
+        assert.isUndefined(sharingUtil.annotationSharingLink(fakeAnnotation));
+      });
     });
 
-    it('returns `html` link if `incontext` link not set on annotation', () => {
-      delete fakeAnnotation.links.incontext;
+    context('sharing an annotation whose document is not on the web', () => {
+      beforeEach(() => {
+        fakeAnnotation.uri = 'file://not-on-web';
+      });
+
+      it('returns `html` link if set on annotation', () => {
+        assert.equal(
+          sharingUtil.annotationSharingLink(fakeAnnotation),
+          'https://www.example2.com'
+        );
+      });
+
+      it('returns `undefined` no `html` link', () => {
+        delete fakeAnnotation.links.html;
 
+        assert.isUndefined(sharingUtil.annotationSharingLink(fakeAnnotation));
+      });
+
+      it('returns `undefined` if no links on annotation', () => {
+        delete fakeAnnotation.links;
+
+        assert.isUndefined(sharingUtil.annotationSharingLink(fakeAnnotation));
+      });
+    });
+  });
+
+  describe('pageSharingLink', () => {
+    it('generates a bouncer link based on the document URI and group id', () => {
       assert.equal(
-        sharingUtil.shareURI(fakeAnnotation),
-        'https://www.example2.com'
+        sharingUtil.pageSharingLink('https://www.example.com', 'testprivate'),
+        'https://hyp.is/go?url=https%3A%2F%2Fwww.example.com&group=testprivate'
       );
     });
+  });
 
-    it('returns `undefined` if links empty', () => {
-      delete fakeAnnotation.links.incontext;
-      delete fakeAnnotation.links.html;
-
-      assert.isUndefined(sharingUtil.shareURI(fakeAnnotation));
+  describe('isShareableURI', () => {
+    [
+      'http://www.example.com',
+      'http://www.foo.bar',
+      'http://hi',
+      'http:foo',
+      'https://www.foo.bar/baz/ding.html',
+    ].forEach(validURI => {
```

Comment:
We are assuming here that the protocol will always be lower-case. I think this is the case as long as the document URI comes from the browser itself. I wonder what happens if the document were to contain something like `<link rel="canonical href="HTTPS://WWW.FOOBAR.COM/BAZ">` though. Please don't spend too many cycles worrying about this, it's very much an edge case.

Reply:
It would be trivial to make the regexp case-insensitive; good catch.